src/main/java/at/hannibal2/skyhanni/features/misc/GeorgeDisplay.kt

```diff
@@ -0,0 +1,115 @@
+    }
+
+    private fun drawDisplay(lore: List<String>) = buildList<Renderable> {
+        this.add(
+            Renderable.string(
+                "§d§lTaming 60 Cost: §r§d(${
+                    if (config.otherRarities) "cheapest" else "exact"
+                } rarity)"
+            )
+        )
+        var totalCost = 0.0
+        for (line in lore) {
+            neededPetPattern.matchMatcher(line) {
+                val origTierString = group("tier")
+                val origPetString = group("pet")
+                val pet = origPetString.uppercase().replace(" ", "_").removePrefix("FROST_")
+                val originalTier = LorenzRarity.getByName(origTierString.uppercase().replace(" ", "_"))?.id ?: 0
+                val (cheapestTier, petPrice) = findCheapestTier(pet, originalTier)
+                val displayPetString =
+                    if (cheapestTier == originalTier) group("fullThing")
+                    else "${LorenzRarity.getById(cheapestTier)?.formattedName} $origPetString"
+                if (petPrice != -1.0) {
+                    totalCost += petPrice
+                    this@buildList.add(
+                        Renderable.clickAndHover(
+                            text = " §7- $displayPetString§7: §6${petPrice.addSeparators()} coins",
+                            tips = listOf(
+                                "§aClick to run §e/ahs ] $origPetString §ato find it on the Auction House.",
+                                "§aNotes: §eSet the rarity filter yourself. §cBooster Cookie required!"
+                            ),
+                            onClick = { LorenzUtils.sendCommandToServer("ahs ] $origPetString") }
+                        ))
+                } else {
+                    this@buildList.add(
+                        Renderable.clickAndHover(
+                            text = " §7- $displayPetString§7: §eNot on AH; view its Wiki article here.",
+                            tips = listOf("§4Click to run §e/wiki $pet §4to view how to obtain it."),
+                            onClick = { LorenzUtils.sendCommandToServer("wiki $pet") }
+                        ))
+                }
+            }
+        }
+        this.add(Renderable.string("§dTotal cost §7(§6Lowest BIN§7): §6${totalCost.addSeparators()} coins"))
+        if (config.otherRarities) this.add(Renderable.string("§c§lDisclaimer:§r§c Total does not include costs to upgrade via Kat."))
+    }
+
+    private fun findCheapestTier(pet: String, originalTier: Int) = buildMap<Int, Double> {
+        this[originalTier] = petInternalName(pet, originalTier).getPetPrice()
+        if (config.otherRarities || this[originalTier] == -1.0) {
+            this[originalTier - 1] = petInternalName(pet, originalTier - 1).getPetPrice(otherRarity = true)
+            if (originalTier != 5) this[originalTier + 1] = petInternalName(pet, originalTier + 1).getPetPrice(otherRarity = true)
+        }
+    }.minBy { it.value }
```

Comment:
No need to be a map (I know I've made the misstake)
```suggestion
    private fun findCheapestTier(pet: String, originalTier: Int) = buildList<Pair<Int, Double>> {
        this.add(originalTier to petInternalName(pet, originalTier).getPetPrice())
        if (config.otherRarities || this[originalTier] == -1.0) {
            this.add(originalTier - 1 to petInternalName(pet, originalTier - 1).getPetPrice(otherRarity = true))
            if (originalTier != 5) this.add(originalTier + 1 to petInternalName(pet, originalTier + 1).getPetPrice(otherRarity = true))
        }
    }.minBy { it.second}
```